### PR TITLE
add ETLS(Enhanced Transport Layer Security)

### PR DESCRIPTION
**Pull request by auxten**
1. add package doc, few code restruct
2. add ETLS(Enhanced Transport Layer Security) to rpc, route test case
3. rename package sign to signature

### CODE REVIEW

File: crypto/hash/hashfuncs.go [hunk 1]
```diff
@@ -22,6 +22,11 @@ import (
 	"hash/fnv"
 )
 
+type HashSuite struct {
```
Comment (covenantdb) [inline crypto/hash/hashfuncs.go:25]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported type HashSuite should have comment or be unexported

Comment (covenantdb) [inline crypto/hash/hashfuncs.go:25]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
type name will be used as hash.HashSuite by other packages, and that stutters; consider calling this Suite

File: crypto/transport/encrypt.go [hunk 2]
```diff
@@ -19,68 +19,65 @@ package transport
...
-	h := md5.New()
-	h.Write(d)
-	return h.Sum(nil)
-}
+	"github.com/thunderdb/ThunderDB/crypto/hash"
+)
 
-func evpBytesToKey(password string, keyLen int) (key []byte) {
-	const md5Len = 16
+// KeyDerivation, according to ANSI X9.63 we should do a key derivation before using
```
Comment (covenantdb) [inline crypto/transport/encrypt.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported function KeyDerivation should be of the form "KeyDerivation ..."

File: crypto/etls/encrypt.go [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"io"
+
+	ec "github.com/btcsuite/btcd/btcec"
+	"github.com/thunderdb/ThunderDB/crypto/hash"
+)
+
+// KeyDerivation, according to ANSI X9.63 we should do a key derivation before using
```
Comment (covenantdb) [inline crypto/etls/encrypt.go]: **[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
comment on exported function KeyDerivation should be of the form "KeyDerivation ..."

**Review by leventeliu — COMMENTED** [review]
We'd better use the third-person singular form of the verb in comments for GoDoc.

**Review by leventeliu — APPROVED** [review]
LGTM, approve.

**Pull request merged**